src/Workspaces/CSharp/Portable/Extensions/ITypeSymbolExtensions.TypeSyntaxGeneratorVisitor.cs

```diff
@@ -241,10 +247,20 @@ public override TypeSyntax VisitNamespace(INamespaceSymbol symbol)
                 }
             }
 
+            private TypeSyntax AddGlobalAlias(INamespaceOrTypeSymbol symbol, SimpleNameSyntax syntax)
+            {
+                return AddInformationTo(
+                    SyntaxFactory.AliasQualifiedName(
+                        CreateGlobalIdentifier(),
```

Comment:
Indeed it looks like we have a bug here in Update 3. If you have System.Xml added with _only_ xml as an extern alias, then:

```
extern alias xml;

namespace ConsoleApplication1
{
    class Program
    {
        static void Main(string[] args)
        {
            XmlDocument foo;
        }
    }
}
```

Invoking add using on XmlDocument offers to add "using System.Xml", which doesn't work. We need xml::System.Xml.

I'm not sure if that's easy to fix (or something we honestly even care about).